http/util.go

```diff
@@ -74,28 +74,28 @@ var (
 	adjustRequest = func(c *vault.Core, r *http.Request) (*http.Request, int) {
 		u := r.URL
 		p := u.Path[3:]
+
+		nsHeaderPath := namespace.Canonicalize(r.Header.Get(consts.NamespaceHeaderName))
 		for _, api := range restrictedAPIs {
-			if strings.HasSuffix(p, api) {
-				return r, 0
+			if strings.HasSuffix(p, api) && (p != "/"+api || nsHeaderPath != "") {
+				c.Logger().Error("got request to restricted API with namespace", "p", p, "api", api, "nsPath", nsHeaderPath)
```

Comment:
```suggestion
				c.Logger().Error("got request to restricted API with namespace", "path", p, "api", api, "nsPath", nsHeaderPath)
```
Just a nit, just "p" is not that clear reading the logs not knowing the underlying code.

Reply:
Ah, that was a debug line, I will remove!